docs/rfc/0003-collaborative-objects.md

```diff
@@ -0,0 +1,506 @@
+the schema we discard the change and all dependent changes. Finally, we have an
+authenticated document which respects a given schema.
+
+It is important to note that this merging is at this point not stored in the
+repository - it can be performed in memory and may be cached. The result is
+that the user sees a single merged view of the object based on the contents
+of the remotes they have replicated. That is, there is no additional
+merge-then-commit step.
+
+### Fetching Collaborative Objects
+
+Each time a repository creates a collaborative object tree it creates a ref
+pointing to that object at `refs/namespaces/<project>/rad/collaborative-objects/<typename>/<object ID>`, 
+where `object ID` is a unique identifier generated at creation time. 
+
+This allows us to fetch subsets of collaborative objects by specifying refspecs
+that match them. The downside is that we are adding a ref to the initial
+advertised refs, each of theses refs is around 250 bytes. If we consider a popular
+repository such as https://github.com/facebook/react/ you can see that they have
+nearly 10,000 issues (including both Open and Closed, which we must). That would 
+mean that the initial ref advertisment in any replication of this repository
+would be ~2.5Mb. If we could use v2 of the git pack protocol this problem would
+go away but alas, we must use v1. 
+
+To get around the problem we can add a parameter to the radicle URL which
+indicates either a single object ID or a type name which we wish to fetch
+from the remote, this would then be passed as a custom header to the transport
+and interpreted on the server. In this manner we can allow clients to choose
+when they want to replicate collaborative objects, which would allow staged
+fetches where we first fetch the repository and identities, and then fetch 
+collaborative objects.
+
+Therefore we add two parameters to a radicle URN:
+
+- `collab_object_id`: specifies that a server should only consider references to the given object ID
+- `collab_type_name`: specifies that a server should only consider references to the given type name
+
+These parameters are emitted by the radicle p2p transport as headers in a 
+similar fashion to the `n=` parameter for the nonce or the `ls=true`
+parameter for selecting the `upload-pack-ls` service. The Radicle git server
+can then use these parameters to filter the refs it operates on.
+
+
+### Updating objects
+
+To make a change to an object we load the existing messages for an object. The 
+application developer provides us with the binary representation of the change
+to that object. We apply the change and ensure that the new object state still
+matches the object schema. At this point the state of the object may depend on
+many contributions from the tracking graph - not just the ones in our own view
+of the project. We now create a commit with our new change in it, referencing
+all the commits containing the direct dependencies of the change as parents.
```

Comment:
Now when you say "all the commits containing the direct dependencies", does that mean we can "reference" other peer's commits? For example, is replying to a comment adding that comment's respective commit to the new commit as a parent?

Reply:
Yep, exactly.

Reply:
Cool, so there is a way that I might see your comment if I'm not tracking you nor the maintainer replies, since someone I'm tracking might reply to you :)